plugin/plugin.go

```diff
@@ -363,6 +373,231 @@ func RegisterShutdown(name string, s Shutter) error {
 	return nil
 }
 
+// Configuration defines the interface for complex_configure callbacks, i.e. Go
+// structs that can hold and validate a configuration
+type Configuration interface {
+	Validate() error
+}
+
+var config_target *Configuration
+var config_targets map[string]*Configuration
```

Comment:
Looks like you're right; this is a hangover from my early experimentation which I had incorrectly believed to be required.